runtime/bcverify/rtverify.c

```diff
@@ -1397,7 +1397,12 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
 
 			receiver = BCV_BASE_TYPE_NULL; /* makes class compare work with statics */
 
-			if (bc & 1) {
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			if ((bc & 1) || (bc == JBwithfield))
+#else /* J9VM_OPT_VALHALLA_VALUE_TYPES */
+			if (bc & 1)
+#endif /* J9VM_OPT_VALHALLA_VALUE_TYPES */
+			{
```

Comment:
We could go into this block if it is JBwithfield, not only JBputfield/JBpustatic now. Please update the comment in the below line.